integration_tests/src/main/python/array_test.py

```diff
@@ -331,4 +331,31 @@ def do_it(spark):
 
     assert_gpu_and_cpu_are_equal_collect(do_it, conf= {
         'spark.sql.legacy.followThreeValuedLogicInArrayExists' : threeVL,
-    })
\ No newline at end of file
+    })
+
+
+array_zips_gen = array_gens_sample + [ArrayGen(map_string_string_gen[0], max_length=5)]
+
+
+@pytest.mark.parametrize('data_gen', array_zips_gen, ids=idfn)
+def test_arrays_zip(data_gen):
+    gen = StructGen(
+        [('a', data_gen), ('b', data_gen), ('c', data_gen), ('d', data_gen)], nullable=False)
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, gen).selectExpr(
+            'arrays_zip(a, b, c, d)',
+            'arrays_zip(a, b, c)',
+            'arrays_zip(a, a)',
```

Comment:
Can we have some tests that mix in scalar arrays too?  Not null but things like

`array()` for an empty array or `array(9, 8, 7, 6)` for a scalar array with the given values?  That would also let us test that when we don't have a named expression that we generate the proper index name of the struct field.  Everything that you have here are named expressions.

Reply:
Added cases for scalar arrays.